pkg/webui/console/components/dev-eui-component/index.js

```diff
@@ -0,0 +1,183 @@
+
+const DevEUIComponent = props => {
+  const { values, setFieldValue, initialValues, devEUISchema } = props
+  const dispatch = useDispatch()
+  const appId = useSelector(selectSelectedApplicationId)
+  const promisifiedIssueDevEUI = attachPromise(issueDevEUI)
+  const fetchDevEUICounter = attachPromise(getApplicationDevEUICount)
+  const euiInputRef = React.useRef(null)
+  const [devEUIGenerated, setDevEUIGenerated] = React.useState(false)
+  const [errorMessage, setErrorMessage] = React.useState(undefined)
+  const applicationDevEUICounter = useSelector(selectApplicationDevEUICount)
+
+  const indicatorContent = Boolean(errorMessage)
+    ? errorMessage
+    : {
+        ...sharedMessages.used,
+        values: {
+          currentValue: applicationDevEUICounter,
+          maxValue: env.devEUIConfig.applicationLimit,
+        },
+      }
+
+  const indicatorCls = classnames(style.indicator, {
+    [style.error]:
+      applicationDevEUICounter === env.devEUIConfig.applicationLimit || Boolean(errorMessage),
+  })
+
+  const handleDevEUIRequest = React.useCallback(async () => {
+    const result = await dispatch(promisifiedIssueDevEUI(appId))
+    await dispatch(fetchDevEUICounter(appId))
+    return result.dev_eui
+  }, [appId, dispatch, fetchDevEUICounter, promisifiedIssueDevEUI])
+
+  const handleGenerate = React.useCallback(async () => {
+    try {
+      const result = await handleDevEUIRequest()
+      setDevEUIGenerated(true)
+      euiInputRef.current.focus()
+      setErrorMessage(undefined)
+      return result
+    } catch (error) {
+      if (error.details[0].name === 'global_eui_limit_reached') {
+        setErrorMessage(sharedMessages.devEUIBlockLimitReached)
+      } else setErrorMessage(sharedMessages.unknownError)
+      setDevEUIGenerated(true)
+    }
+  }, [handleDevEUIRequest])
+
+  const generateDeviceId = React.useCallback(
+    (device = {}) => {
+      const { ids: idsValues = {} } = device
+
+      try {
+        devEUISchema.validateSync(idsValues.dev_eui)
+        return `eui-${idsValues.dev_eui.toLowerCase()}`
+      } catch (e) {
+        // We dont want to use invalid `dev_eui` as `device_id`.
+      }
+
+      return initialValues.ids.device_id || undefined
+    },
+    [initialValues.ids.device_id, devEUISchema],
+  )
+
+  const handleIdPrefill = React.useCallback(() => {
+    if (values) {
+      // Do not overwrite a value that the user has already set.
+      if (values.ids.device_id === initialValues.ids.device_id) {
+        const generatedId = generateDeviceId(values)
+        setFieldValue('ids.device_id', generatedId)
+      }
+    }
+  }, [values, setFieldValue, initialValues, generateDeviceId])
+
+  const devEUIGenerateDisabled =
+    applicationDevEUICounter === env.devEUIConfig.applicationLimit ||
+    !env.devEUIConfig.devEUIIssuingEnabled ||
+    devEUIGenerated
+
+  return env.devEUIConfig.devEUIIssuingEnabled ? (
+    <Form.Field
+      title={sharedMessages.devEUI}
+      name="ids.dev_eui"
```

Comment:
```suggestion
      name={props.name}
```
This way, you can pass the relevant field name `authenticated_identifiers.dev_eui` / `ids.dev_eui` based on whether you are claiming or not.